Mathlib/Data/List/Destutter.lean

```diff
@@ -61,6 +64,15 @@ theorem destutter'_singleton : [b].destutter' R a = if R a b then [a, b] else [a
   split_ifs with h <;> simp! [h]
 #align list.destutter'_singleton List.destutter'_singleton
 
+/-- `destutter'` always has length at least 1. -/
+theorem length_destutter'_pos : 0 < (destutter' R a l).length := by
```

Comment:
```suggestion
theorem destutter'_ne_nil : destutter' R a l ≠ [] = by
```
is the more canonical spelling

Reply:
Alright! So it turns out `destutter'_ne_nil` already exists so I just changed my one usage,
`exact length_destutter'_pos _ _`
into
`List.length_pos_of_ne_nil (destutter'_ne_nil _ _)`